libs/model/src/aggregates/user/signIn/utils.ts

```diff
@@ -136,6 +182,15 @@ export async function findOrCreateUser({
   if (privyUserId && foundUser && !signedInUser && !foundUser.privy_id)
     await updatePrivyId(foundUser.id!);
 
+  const tier = privyUserId
```

Comment:
There is actually a pre-existing bug here. The `privyUserId` may be defined even when not using SSO (i.e. sign in with native wallet through Privy). As such, tier should be `UserTierMap.SocialVerified` IFF `privyUserId` AND `ssoInfo` is defined. That means this condition to check for native wallet balance should happen when `privyUserId `is defined AND `ssoInfo` is undefined.

I also think the `checkNativeWalletBalance` should only return the tier (not update the user) and then we should update both privyId and tier in the same User update query.

Reply:
can we have a case where privyUserId is undefined? that's the only edge case that can break the suggestion above I think

Reply:
I was talking about reusing the existing update function but I already pushed something that might work